arangod/Aql/AqlItemBlock.cpp

```diff
@@ -575,34 +584,38 @@ auto AqlItemBlock::slice(arangodb::containers::SmallVector<std::pair<size_t, siz
   }
 
   arangodb::containers::HashSet<void*> cache;
-  cache.reserve(numRows * _nrRegs / 4 + 1);
+  cache.reserve(numRows * _numRegisters / 4 + 1);
 
-  SharedAqlItemBlockPtr res{_manager.requestBlock(numRows, _nrRegs)};
+  SharedAqlItemBlockPtr res{_manager.requestBlock(numRows, _numRegisters)};
   size_t targetRow = 0;
   for (auto const& [from, to] : ranges) {
+    size_t const effectiveTo = std::min<size_t>(to, _numEffectiveRows);
+
     for (size_t row = from; row < to; row++, targetRow++) {
-      for (RegisterId col = 0; col < _nrRegs; col++) {
-        ::copyValueOver(cache, _data[getAddress(row, col)], targetRow, col, res);
+      if (row < effectiveTo) {
+        for (RegisterId col = 0; col < _numRegisters; col++) {
+          ::copyValueOver(cache, _data[getAddress(row, col)], targetRow, col, res);
+        }
       }
       res->copySubqueryDepthFromOtherBlock(targetRow, *this, row, false);
```

Comment:
The shadowRowDepth is aligned with the row data.
So to my best knowledge it is also fine to just copy it, if row < effectiveTo otherwise there will be no data anyways.
In this case we can again reduce the loop to stop at effectiveTo instead of to.